asab/api/discovery.py

```diff
@@ -28,17 +28,26 @@ def __init__(self, app, zkc, service_name="asab.DiscoveryService") -> None:
 		self._cache_lock = asyncio.Lock()
 		self._ready_event = asyncio.Event()
 
+		self._concurrent_future = None
```

Comment:
The original idea was to pass the future to the TaskService, if it is possible:

```
		future = asyncio.run_coroutine_threadsafe(self._rescan_advertised_instances(), self.App.Loop)
		self.App.TaskService.schedule([future])
```

In this case you would not need to care about the await of the task in finalize.

Reply:
:( to nefunguje